.github/workflows/rerun-workflows.yml

```diff
@@ -0,0 +1,31 @@
+name: Rerun Flaky Workflows
+
+on:
+  workflow_run:
+    workflows: [Backend, Driver Tests, E2E Tests, Frontend]
+    types: [completed]
+    branches: [master, 'release-**']
+
+jobs:
+  rerun-on-failure:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.workflow_run.conclusion }} == 'failure'
+    steps:
+      - uses: actions/github-script@v6
+        with:
+          script: |
+            const MAX_ATTEMPTS = 2;
+            const ATTEMPT = ${{ github.event.workflow_run.run_attempt }};
+
+            if (ATTEMPT <= MAX_ATTEMPTS) {
+              console.log("Rerruning...");
+
+              github.rest.actions.reRunWorkflowFailedJobs({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                run_id: ${{ github.event.workflow_run.id }},
+              });
```

Comment:
This will rerun only failed jobs from a workflow:
https://docs.github.com/en/rest/actions/workflow-runs#re-run-failed-jobs-from-a-workflow-run